### ISSUE TITLE
[BUG] Cannot delete statistic cache file

### ISSUE THREAD

**Issue by ozzy1873** [JaffaKetchup/flutter_map_tile_caching#73]
### What is the bug?

Exception occurs when deleting attempting to delete a file. This happened after upgrading Flutter to 3.3.7 and running in debug mode. 

I/flutter (22291): ----------------FIREBASE CRASHLYTICS----------------
I/flutter (22291): FileSystemException: Cannot delete file, path = '/data/user/0/com.mycompany.myapp/app_flutter/fmtc/stores/OsmStoreDirectory/stats/length.cache' (OS Error: No such file or directory, errno = 2)
I/flutter (22291): 
I/flutter (22291): #0      _File._delete.<anonymous closure> (dart:io/file_impl.dart:283:9)
I/flutter (22291): #1      _rootRunUnary (dart:async/zone.dart:1399:47)
I/flutter (22291): #2      _CustomZone.runUnary (dart:async/zone.dart:1300:19)
I/flutter (22291): <asynchronous suspension>
I/flutter (22291): #3      Future.wait.<anonymous closure> (dart:async/future.dart:522:21)
I/flutter (22291): <asynchronous suspension>
I/flutter (22291): ----------------------------------------------------

### What is the expected behaviour?

No exception should happen. I can catch the error in my root zone and log it, so this is not a showstopper for me.

### How can we reproduce this issue?

```dart
This happened after upgrading Flutter from 3.3.6 to 3.3.7, and then running my app in debug mode. I saw this also after the previous update but did not report it then.
```

### Do you have a potential solution?

Don't know, maybe check for the file/directory existence before attempting to delete?

### Platforms Affected

Android

### Severity

Minimum: Allows normal functioning

### Frequency

Rarely: Occurs every so often

### Requirements

- [X] I agree to follow this project's [Code of Conduct](https://github.com/jaffaketchup/flutter_map_tile_caching/blob/main/CODE_OF_CONDUCT.md)
- [X] My Flutter/Dart installation is unaltered, and `flutter doctor` finds no relevant issues
- [X] I am using the [latest stable version](https://pub.dev/packages/flutter_map_tile_caching) of this package
- [X] I have checked the FAQs section on the documentation website
- [X] I have checked for similar issues which may be duplicates

**Comment by JaffaKetchup** [comment]
Hi @ozzy1873. What were you doing that caused that error - it looks like you were invalidating cached statistics, but I might be wrong? Additionally, can I ask which version you are on - I have a feeling I fixed a very similar issue before?

**Comment by ozzy1873** [comment]
When the error occurred, I started the app, which calls this:

FlutterMapTileCaching.initialise(await RootDirectory.normalCache);

Then I displayed the map, which I think is when the error occurred. I did not explicitly invalidate the cache. 

Versions:

flutter_map: ^3.0.0
flutter_map_tile_caching: ^6.2.0

**Comment by JaffaKetchup** [comment]
Ok, that is strange, I'll have a look into it. Thanks for the report.

**Comment by caini1213** [comment]
> When the error occurred, I started the app, which calls this:
> 
> FlutterMapTileCaching.initialise(await RootDirectory.normalCache);
> 
> Then I displayed the map, which I think is when the error occurred. I did not explicitly invalidate the cache.
> 
> Versions:
> 
> flutter_map: ^3.0.0 flutter_map_tile_caching: ^6.2.0

Ya, I got the same problem too which keep me downloading failed.

**Comment by JaffaKetchup** [comment]
Can one of you send over your 'fmtc' directory if you can (without an excessive number of tiles)? Should make it easier for me to reproduce.

**Comment by caini1213** [comment]
https://drive.google.com/drive/folders/1oAdxINoUv4F1KX5QFYgTGOIxfUXUuD6u?usp=share_link

**Comment by JaffaKetchup** [comment]
Unfortunately, I cannot reproduce this. Is there a bigger stack trace @ozzy1873 can send? Note that at one point I silence and ignore a deletion of this file because this action's failure isn't really an issue (see the screenshot), but this shouldn't get through to Crashlytics, and it certainly shouldn't stop a download @caini1213.

![image](https://user-images.githubusercontent.com/58115698/201201925-9decfff5-ebaa-423d-92de-38562ec8a0c1.png)

**Comment by ozzy1873** [comment]
Unfortunately, that is the only stack trace I have.

**Comment by JaffaKetchup** [comment]
Well, I'll keep an eye on it for the next release, but without a longer stack trace that actually tells me what line is doing the deletion, there's not much I can do (I can't see anything wrong). Thanks for the report!

**Comment by JaffaKetchup** [comment]
Hey @ozzy1873,

Can you try manually depending on the '[next-version](https://github.com/JaffaKetchup/flutter_map_tile_caching/tree/next-version)' branch, to see if that resolves this issue? Try replacing the current dependency with this:
```yaml
flutter_map_tile_caching:
  git:
    url: https://github.com/JaffaKetchup/flutter_map_tile_caching.git
    ref: next-version
``` 

Hoping to hear from you!

**Comment by caini1213** [comment]
Unfortunately, it's still the same for my side.

**Comment by JaffaKetchup** [comment]
@caini1213 Is there any way you can verify that the code actually changed on your system? If you 'View Definition' of any one of the symbols from this library, it should show the breadcrumbs to the file somewhere in the IDE. Can you check that the breadcrumbs reference git instead of pub.dev?

**Comment by JaffaKetchup** [comment]
This will be fixed by v7 (#78).

**Comment by sikandernoori** [comment]
following this issue
